FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -172,25 +181,157 @@ final class ComposeMessageService {
         } catch {
             throw ComposeMessageError.gatewayError(error)
         }
     }
+
+    // MARK: - Message password
+    func composePasswordMessage(from message: SendableMsg) async throws -> CoreRes.ComposeEmail {
+        let replyToken = try await enterpriseServer.getReplyToken(for: message.from)
+
+        let messageUrl = try await prepareAndUploadPwdEncryptedMsg(
+            message: message,
+            replyToken: replyToken
+        )
+
+        let html = generatePasswordMessageHtml(sender: message.from, url: messageUrl)
+
+        let encryptedTextFile = try await core.encryptMsg(msg: message, fmt: .encryptInline)
+        let bodyAttachment = SendableMsg.Attachment(
+            name: "encrypted.asc",
+            type: "application/pgp-encrypted",
+            base64: encryptedTextFile.mimeEncoded.base64EncodedString()
+        )
+
+        var encryptedAttachments: [SendableMsg.Attachment] = []
+
+        for attachment in message.atts {
+            guard let data = Data(base64Encoded: attachment.base64) else { continue }
+
+            let encryptedFile = try await core.encryptFile(
+                pubKeys: message.pubKeys,
+                fileData: data,
+                name: attachment.name
+            )
+            let encryptedAttachment = SendableMsg.Attachment(
+                name: "\(attachment.name).pgp",
+                type: "application/pgp-encrypted",
+                base64: encryptedFile.encryptedFile.base64EncodedString()
+            )
+            encryptedAttachments.append(encryptedAttachment)
+        }
+
+        let passwordMessage = SendableMsg(
+            text: html,
+            html: html,
+            to: message.to,
+            cc: message.cc,
+            bcc: message.bcc,
+            from: message.from,
+            subject: message.subject,
+            replyToMimeMsg: message.replyToMimeMsg,
+            atts: [bodyAttachment] + encryptedAttachments,
+            pubKeys: nil,
+            signingPrv: nil,
+            password: nil
+        )
+        return try await core.composeEmail(msg: passwordMessage, fmt: .plain)
+    }
+
+    func prepareAndUploadPwdEncryptedMsg(message: SendableMsg, replyToken: String) async throws -> String {
+        let formatted = try await formatPasswordMessage(
+            message: message,
+            replyToken: replyToken
+        )
+
+        let formattedMessage = SendableMsg(
+            text: formatted.mimeEncoded.toStr(),
+            html: formatted.mimeEncoded.toStr(),
+            to: message.to,
+            cc: message.cc,
+            bcc: message.bcc,
+            from: message.from,
+            subject: message.subject,
+            replyToMimeMsg: message.replyToMimeMsg,
+            atts: message.atts,
+            pubKeys: [],
+            signingPrv: nil,
+            password: message.password
+        )
+
+        let encoded = try await core.encryptMsg(msg: formattedMessage, fmt: .encryptInline)
```

Comment:
I think there's a step missing there. The `encoded` value here should be a plain-encoded mime message. Then the next step will be to encrypt it. So it will be something like:

```swift
let mimeEncodedPlainWithAttachmeents = try await core.composeEmail(sendableMsg, Type.plain)
let encrypted = try await core.encryptMsg(data: mimeEncodedPlainWithAttachmeents, fmt: .encryptInline)
```

Else the recipient won't be able to open attachments on the web portal - they will be missing.

Reply:
Probably it's a bit confusing naming - I tested emails with attachments, they work well - attachments are visible on web portal after entering message password.
I'll improve readability here